packages/apollo-engine-reporting/src/__tests__/extension.test.ts

```diff
@@ -262,189 +266,222 @@ describe("variableJson output for sendVariableValues transform: custom function
-  ).toEqual(JSON.stringify("[Unable to convert value to JSON]"));
+    makeTraceDetails(circularVariables, { all: true }).variablesJson['bad'],
+  ).toEqual(JSON.stringify('[Unable to convert value to JSON]'));
 });
 
 function makeTestHTTP(): Trace.HTTP {
   return new Trace.HTTP({
     method: Trace.HTTP.Method.UNKNOWN,
     host: null,
-    path: null
+    path: null,
   });
 }
 
 /**
  * TESTS FOR THE sendHeaders REPORTING OPTION
  */
 const headers = new Headers();
-headers.append("name", "value");
-headers.append("authorization", "blahblah"); // THIS SHOULD NEVER BE SENT
+headers.append('name', 'value');
+headers.append('authorization', 'blahblah'); // THIS SHOULD NEVER BE SENT
 
-const headersOutput = { name: new Trace.HTTP.Values({ value: ["value"] }) };
+const headersOutput = { name: new Trace.HTTP.Values({ value: ['value'] }) };
 
-describe("tests for the shouldReportQuery reporting option", () => {
-  it("report no traces", async () => {
+describe('tests for the shouldReportQuery reporting option', () => {
+  it('report no traces', async () => {
     const schema = makeExecutableSchema({ typeDefs });
     addMockFunctionsToSchema({ schema });
     enableGraphQLExtensions(schema);
 
     async function addTrace(_args: AddTraceArgs) {
-      throw new Error("Should not add any traces");
+      throw new Error('Should not add any traces');
     }
 
     const reportingExtension = new EngineReportingExtension(
       {
         shouldReportQuery: () => {
           return false;
-        }
+        },
       },
       addTrace,
-      "schema-hash"
+      'schema-hash',
     );
 
     const stack = new GraphQLExtensionStack([reportingExtension]);
     const requestDidEnd = stack.requestDidStart({
-      request: new Request("http://localhost:123/foo"),
+      request: new Request('http://localhost:123/foo'),
       queryString: query,
       requestContext: {
         request: {
           query,
-          operationName: "q",
+          operationName: 'q',
           extensions: {
-            clientName: "testing suite"
-          }
+            clientName: 'testing suite',
+          },
         },
         context: {},
-        cache: new InMemoryLRUCache()
+        cache: new InMemoryLRUCache(),
       },
-      context: {}
+      context: {},
     });
 
     await graphql({
       schema,
       source: query,
-      contextValue: { _extensionStack: stack }
+      contextValue: { _extensionStack: stack },
     });
 
     requestDidEnd();
   });
 
-  it("report traces based on operation name", async () => {
+  it('report traces based on operation name', async () => {
     const schema = makeExecutableSchema({ typeDefs });
     addMockFunctionsToSchema({ schema });
     enableGraphQLExtensions(schema);
 
-    async function addTrace(args: AddTraceArgs) {
-      expect(args.operationName).toEqual("report");
+    let tracesAdded = 0;
+
+    async function addTrace(_args: AddTraceArgs) {
+      tracesAdded += 1;
```

Comment:
Previous mock comment applies, but here you have the added benefit of it keeping track of the call count for you.

```js
expect(addTrace).toBeCalledTimes(1);
```

You can also test _what_ it was called with if you'd like to enforce that:

```js
expect(addTrace).toBeCalledWith(/* expected arguments */);
```